crates/node-core/src/exit.rs

```diff
@@ -23,10 +21,10 @@ pub struct NodeExitFuture {
 impl NodeExitFuture {
     /// Create a new `NodeExitFuture`.
     pub const fn new(
-        consensus_engine_rx: oneshot::Receiver<Result<(), BeaconConsensusEngineError>>,
+        consensus_engine_fut: Pin<Box<dyn Future<Output = eyre::Result<()>>>>,
```

Comment:
we can relax this and instead require that this is `F: Future + 'static + Send` and Box::pin in this function